src/data-models/context/todo-context.ts

```diff
@@ -0,0 +1,38 @@
+import { createContext } from "react";
+
+
+
+export interface ITodoListContext {
+    Provider?: React.Provider<ITodoListContext>;
+    removeTodo(id: string): void;
+    toggleTodo(id: string, completed: boolean): void;
+}
+
+
+
+export const TodoContextController = (() => {
+
+    let context: React.Context<ITodoListContext> | undefined;
+
+
+
+    function initializeContext() {
+        return createContext(undefined);
+    }
+
+
+
+    function getContext() {
+
+        if (!context)
+        {
+            context = initializeContext();
```

Comment:
how about we directly call `createContext` here? can't see a reason to wrap it in another function